llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -12583,16 +12583,37 @@ struct AAAddressSpaceImpl : public AAAddressSpace {
   }
 
   ChangeStatus updateImpl(Attributor &A) override {
+    assert(A.getInfoCache().getFlatAddressSpace().has_value());
```

Comment:
This assert is redundant, the attempt to use it will fail anyway (plus I still think it's nicer to not use optional, and just let the invalid address space value naturally not do anything)